src/main/java/org/openrewrite/java/migrate/AddStaticVariableOnProducerSessionBean.java

```diff
@@ -66,13 +99,24 @@ private boolean isInSessionBean() {
                             return false;
                         }
                         return hasAnnotation(parentClass, "@jakarta.ejb.Singleton") ||
-                                hasAnnotation(parentClass, "@jakarta.ejb.Stateful") ||
-                                hasAnnotation(parentClass, "@jakarta.ejb.Stateless");
+                               hasAnnotation(parentClass, "@jakarta.ejb.Stateful") ||
+                               hasAnnotation(parentClass, "@jakarta.ejb.Stateless");
+                    }
+
+                    private boolean isInXml() {
+                        J.ClassDeclaration parentClass = getCursor().firstEnclosing(J.ClassDeclaration.class);
+                        if (parentClass != null && parentClass.getType() instanceof JavaType.FullyQualified) {
```

Comment:
`instanceof` does a null check already.
```suggestion
                        if (parentClass.getType() instanceof JavaType.FullyQualified) {
```